android/src/main/java/bz/rxla/flutter/speechrecognition/SpeechRecognitionPlugin.java

```diff
@@ -58,7 +59,12 @@ public void onMethodCall(MethodCall call, Result result) {
             case "speech.activate":
                 // FIXME => Dummy activation verification : we assume that speech recognition permission
                 // is declared in the manifest and accepted during installation ( AndroidSDK 21- )
-                Locale locale = activity.getResources().getConfiguration().locale;
+                Locale locale;
+                if (android.os.Build.VERSION.SDK_INT >= android.os.Build.VERSION_CODES.N) {
+                    locale = activity.getResources().getConfiguration().getLocales().get(0);
```

Comment:
How about moving the part `activity.getResources().getConfiguration()` outside of conditional statements to avoid code repetition? We would have something like:

```
Configuration config = activity.getResources().getConfiguration();
Locale locale;
if (android.os.Build.VERSION.SDK_INT >= android.os.Build.VERSION_CODES.N) {
    locale = config.getLocales().get(0);
} else {
    locale = config.locale;
}
```

Reply:
Sure, good point. Done.